cirq-core/cirq/experiments/qubit_characterizations.py

```diff
@@ -213,21 +213,73 @@ def single_qubit_randomized_benchmarking(
         A RandomizedBenchMarkResult object that stores and plots the result.
     """
 
+    qubits = cast(Iterator['cirq.Qid'], (qubit,))
+    result = parallel_single_qubit_randomized_benchmarking(
+        sampler,
+        qubits,
+        use_xy_basis,
+        num_clifford_range=num_clifford_range,
+        num_circuits=num_circuits,
+        repetitions=repetitions,
+    )
+    return result[qubit]
+
+
+def parallel_single_qubit_randomized_benchmarking(
+    sampler: 'cirq.Sampler',
+    qubits: Iterator['cirq.Qid'],
+    use_xy_basis: bool = True,
+    *,
+    num_clifford_range: Sequence[int] = tuple(np.logspace(np.log10(5), 3, 5, dtype=int)),
+    num_circuits: int = 10,
+    repetitions: int = 600,
+) -> dict:
+    """Clifford-based randomized benchmarking (RB) single qubits in parallel.
+
+    This is the same as `single_qubit_randomized_benchmarking` except on all
+    of the specified qubits in parallel, i.e. with the individual randomized
+    benchmarking circuits zipped together.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        use_xy_basis: Determines if the Clifford gates are built with x and y
+            rotations (True) or x and z rotations (False).
+        qubits: The qubits to benchmark.
+        num_clifford_range: The different numbers of Cliffords in the RB study.
+        num_circuits: The number of random circuits generated for each
+            number of Cliffords.
+        repetitions: The number of repetitions of each circuit.
+
+    Returns:
+        A dictionary from qubits to RandomizedBenchMarkResult objects.
+    """
+
     cliffords = _single_qubit_cliffords()
     c1 = cliffords.c1_in_xy if use_xy_basis else cliffords.c1_in_xz
-    cfd_mats = np.array([_gate_seq_to_mats(gates) for gates in c1])
+    clifford_mats = np.array([_gate_seq_to_mats(gates) for gates in c1])
 
-    gnd_probs = []
-    for num_cfds in num_clifford_range:
-        excited_probs_l = []
+    # create circuits
+    circuits_all = []
+    for num_cliffords in num_clifford_range:
         for _ in range(num_circuits):
-            circuit = _random_single_q_clifford(qubit, num_cfds, c1, cfd_mats)
-            circuit.append(ops.measure(qubit, key='z'))
-            results = sampler.run(circuit, repetitions=repetitions)
-            excited_probs_l.append(np.mean(results.measurements['z']))
-        gnd_probs.append(1.0 - np.mean(excited_probs_l))
+            circuits_all.append(
+                _create_parallel_rb_circuit(qubits, num_cliffords, c1, clifford_mats)
+            )
 
-    return RandomizedBenchMarkResult(num_clifford_range, gnd_probs)
+    # run circuits
+    results_all = sampler.run_batch(circuits_all, repetitions=repetitions)
+    gnd_probs: dict = {q: [] for q in qubits}
+    idx = 0
+    for num_cliffords in num_clifford_range:
+        excited_probs_l: dict = {q: [] for q in qubits}
```

Comment:
nit: I'd suggest omitting the `_l` suffix, which is a bit obscure. Also, a full type annotation would be helpful:
```suggestion
        excited_probs: Dict['cirq.Qid', List[float]] = {q: [] for q in qubits}
```